### PR TITLE
[TASK] Add possibility to forward the reason to the error page

### PR DESCRIPTION
**Pull request by cjanody**
Hi,
We like your extension and made some improvement to forward the reason to the error page.
Why?! If someone tries to access a page which is not available in his language, he may want to know the page exists in another language and will be able to access it.
We work on a plugin to do it and will make another pull request soon.
Thanks for reading.
Regards

### CODE REVIEW

File: Classes/Utility/CustomErrorPageUtility.php [hunk 1]
```diff
@@ -142,7 +142,30 @@ private function findErrorPage(string $currentUrl, array $configuration, int $ty
...
+                    $GLOBALS['TSFE']->tmpl->start($GLOBALS['TSFE']->rootLine);
+                    $GLOBALS['TSFE']->getConfigArray();
+                    $GLOBALS['TSFE']->settingLanguage();
+                    $GLOBALS['TSFE']->calculateLinkVars();
+
+                    if ($GLOBALS['TSFE']->linkVars) {
+                        $url .= $GLOBALS['TSFE']->linkVars;
+                    }
+                } catch (\Exception $e) {
+                    $message = 'Could not build a localized "pageNotFound" that belongs to this domaine. '
```
Comment (bitmotionAE) [inline Classes/Utility/CustomErrorPageUtility.php]: *domain

File: Classes/Utility/CustomErrorPageUtility.php [hunk 2]
```diff
@@ -142,7 +142,30 @@ private function findErrorPage(string $currentUrl, array $configuration, int $ty
...
+                        . "(pid: {$GLOBALS['TSFE']->domainStartPage})";
+                    $this->logger->critical($message);
+                }
+
+                if ($configuration[$hostName]['forward-reason'] === true
+                    || $configuration['_DEFAULT']['forward-reason'] === true) {
+                    $url .= '&reason=' . urlencode($params['reasonText']);
```
Comment (bitmotionAE) [inline Classes/Utility/CustomErrorPageUtility.php]: Is the variable $params['reasonText'] always set? If not, you could add an !empty check before.

File: Classes/Utility/CustomErrorPageUtility.php (hunk 1 above)
Comment (cjanody) [reply Classes/Utility/CustomErrorPageUtility.php]: Done

File: Classes/Utility/CustomErrorPageUtility.php (hunk 2 above)
Comment (cjanody) [reply Classes/Utility/CustomErrorPageUtility.php]: Check added.

**Comment by FlorianWessels** [comment]
Thanks for your work. Good job :)

**Pull request merged**